cms/djangoapps/contentstore/views/course.py

```diff
@@ -374,15 +376,36 @@ def course_filter(course_summary):
         # TODO remove this condition when templates purged from db
         if course_summary.location.course == 'templates':
             return False
-
         return has_studio_read_access(request.user, course_summary.id)
+
+    def active_course_filter(course_summary):
+        """
+        Return accessible, usable courses, filtering out archived courses if requested.
+        """
+        if split_archived and course_summary.has_ended():
+            return False
+        return course_filter(course_summary)
+
+    def archived_course_filter(course_summary):
+        """
+        Return accessible, usable, archived courses
+        """
+        if split_archived and course_summary.has_ended():
+            return course_filter(course_summary)
+        return False
+
     if org is not None:
         courses_summary = [] if org == '' else CourseOverview.get_all_courses(orgs=[org])
     else:
         courses_summary = modulestore().get_course_summaries()
-    courses_summary = six.moves.filter(course_filter, courses_summary)
+
+    active_courses = six.moves.filter(active_course_filter, courses_summary)
+    archived_courses = ()
+    if split_archived:
+        archived_courses = six.moves.filter(archived_course_filter, courses_summary)
     in_process_course_actions = get_in_process_course_actions(request)
-    return courses_summary, in_process_course_actions
+
+    return active_courses, in_process_course_actions, archived_courses
```

Comment:
It would actually be OK to return only the first two if `split_archived` is not set.  You would have needed to modify fewer tests.  Since you've done that now anyway, it doesn't matter much either way.